samples/hello_events.c

```diff
@@ -215,147 +200,140 @@ int find_fpga(fpga_token *fpga, uint32_t *num_matches)
 out_destroy:
 	dres = fpgaDestroyProperties(&filter);
 	ON_ERR_GOTO(dres, out, "destroying properties object");
-	
 out:
 	return (res == FPGA_OK) ? dres: res;
 }
 
 
-/* functions to get the bus number when there are multiple buses */
-struct bdf_info {
-	uint8_t bus;
-};
+/* function to get the bus number when there are multiple buses */
+fpga_result get_bus(fpga_token tok, uint8_t *bus)
+{
+	fpga_result res1 = FPGA_OK;
+	fpga_result res2 = FPGA_OK;
+	fpga_properties props = NULL;
 
-fpga_result get_bus_info(fpga_token tok, struct bdf_info *finfo){
-	fpga_result res = FPGA_OK;
-	fpga_properties props;
-	res = fpgaGetProperties(tok, &props);
-	ON_ERR_GOTO(res, out, "reading properties from Token");
+	res1 = fpgaGetProperties(tok, &props);
+	ON_ERR_GOTO(res1, out, "reading properties from Token");
 
-	res = fpgaPropertiesGetBus(props, &finfo->bus);
-	ON_ERR_GOTO(res, out_destroy, "Reading bus from properties");
+	res1 = fpgaPropertiesGetBus(props, bus);
+	ON_ERR_GOTO(res1, out_destroy, "Reading bus from properties");
 	
 out_destroy:
-	res = fpgaDestroyProperties(&props);
-	ON_ERR_GOTO(res, out, "fpgaDestroyProps");
-
+	res2 = fpgaDestroyProperties(&props);
+	ON_ERR_GOTO(res2, out, "fpgaDestroyProps");
 out:
-	return res;
-}
-
-void print_bus_info(struct bdf_info *info){
-	printf("Running on bus 0x%02X. \n", info->bus);
+	return res1 != FPGA_OK ? res1 : res2;
 }
 
-
-
-
-
-
 int main(int argc, char *argv[])
 {
-	fpga_token         fpga_device_token;
-	fpga_handle        fpga_device_handle;
+	fpga_token         fpga_device_token = NULL;
+	fpga_handle        fpga_device_handle = NULL;
 	uint32_t           num_matches = 1;
-	fpga_result res;
-	fpga_event_handle eh;
-	uint64_t count = 0;
-	pid_t pid;
-	struct pollfd pfd;
-	int timeout = 10000;
-	int poll_ret = 0;
-	ssize_t bytes_read = 0;
-	struct bdf_info info;
-
-	res = parse_args(argc, argv);
-	ON_ERR_GOTO(res, out_exit, "parsing arguments");
-
-
-	res = find_fpga(&fpga_device_token, &num_matches);
-	ON_ERR_GOTO(res, out_exit, "finding FPGA accelerator");
+	fpga_result        res1 = FPGA_OK;
+	fpga_result        res2 = FPGA_OK;
+	fpga_event_handle  eh;
+	uint64_t           count = 0;
+	pid_t              pid;
+	struct pollfd      pfd;
+	int                timeout = 10000;
+	int                poll_ret = 0;
+	ssize_t            bytes_read = 0;
+	uint8_t            bus = 0xff;
+
+	res1 = parse_args(argc, argv);
+	ON_ERR_GOTO(res1, out_exit, "parsing arguments");
+
+	res1 = find_fpga(&fpga_device_token, &num_matches);
+	ON_ERR_GOTO(res1, out_exit, "finding FPGA accelerator");
 
 	if (num_matches < 1) {
-		fprintf(stderr, "No matches for bus number provided \n");
-		res = FPGA_NOT_FOUND;
+		fprintf(stderr, "No matches for bus number provided.\n");
+		res1 = FPGA_NOT_FOUND;
 		goto out_exit;
 	}
 
+	res1 = get_bus(fpga_device_token, &bus);
+	ON_ERR_GOTO(res1, out_exit, "getting bus num");	
```

Comment:
We don't destroy token if we `goto out_exit`

Reply:
Right, the jump should be to out_destroy_tok.